src/bellpepper/r1cs.rs

```diff
@@ -72,15 +71,14 @@ macro_rules! impl_nova_shape {
             &constraint.2,
           );
         }
-
         assert_eq!(num_cons_added, num_constraints);
 
-        let S: R1CSShape<G> = {
-          // Don't count One as an input for shape's purposes.
-          let res = R1CSShape::new(num_constraints, num_vars, num_inputs - 1, &A, &B, &C);
-          res.unwrap()
-        };
+        A.cols = num_vars + num_inputs;
+        B.cols = num_vars + num_inputs;
+        C.cols = num_vars + num_inputs;
 
+        // Don't count One as an input for shape's purposes.
```

Comment:
What is the rationale for this?

Reply:
I believe it's because `R1CSShape.num_vars` is meant to correspond to the size of the witness, as checked here: https://github.com/microsoft/Nova/blob/9165211d352b6e5f9bcaef2e6cc5cb79666ecf6f/src/r1cs.rs#L245 — but the witness doesn't explicitly contain the 'one' value. Rather, it's added in when checking satisfaction: https://github.com/microsoft/Nova/blob/9165211d352b6e5f9bcaef2e6cc5cb79666ecf6f/src/r1cs.rs#L250.

This is just the same code and comment as previously existed but refactored to be less verbose.

Reply:
Got it, sounds good! thanks!